furryfeet_be/furryfeet/views.py

```diff
@@ -104,3 +104,156 @@ def post(self, request, pk):
 
         serializer = UserSerializer(user)
         return Response(serializer.data, status=status.HTTP_200_OK)
+    
+
+class DogWalkerView(APIView):
+    def get(self, request):
+        token = request.COOKIES.get('jwt')
+
+        if not token:
+          raise AuthenticationFailed("Unauthenticated!")
+        try:
+            payload = jwt.decode(token, 'secret', algorithms=['HS256'])
+        except jwt.ExpiredSignatureError:
+            raise AuthenticationFailed("Unauthenticated!")
+         
+        try:
+            dog_walker = DogWalker.objects.get(user=request.user)
```

Comment:
use the id from the payload to get the user from the database and then send that user to this function here. otherwise we get an error. See AddUserImageView for reference.